numpy/compat/py3k.py

```diff
@@ -104,7 +100,11 @@ def is_pathlib_path(obj):
 
     Prefer using `isinstance(obj, os_PathLike)` instead of this function.
     """
-    return Path is not None and isinstance(obj, Path)
+    from pathlib import Path
+    from warnings import warn
+    warn("The use of is_pathlib_path is deprecated. "
```

Comment:
You should use DeprecationWarning here. I also think you should probably just import warnings at the top level.

Reply:
Re DeprecationWarning, do those actually get heard by downstream libraries? I think python started to silence those in many situations.

Reply:
@rgommers had an NEP that touched on this. At any rate, `DeprecationWarning` is currently used throughout NumPy, so we should keep using it.

Reply:
> shall use VisibleDeprecationWarning rather than DeprecationWarning for cases of relevance to end users (as opposed to cases only relevant to libraries building on top of NumPy).

I guess it just depends on when you want to remove those 4 lines from the numpy library.

Reply:
This is barely public API anyway, so I wouldn't worry about it.

Reply:
Also: deprecation warnings need a removal date and release version in a comment on the line above the warning - search the source for `DeprecationWarning` to see an example

Reply:
> This is barely public API anyway, so I wouldn't worry about it.

I was going to say that, but i did a quick search on github and some people seem to be using it internally. Gotta love how ubiquitous numpy is :D.

Reply:
My comment about the deprecation date going in the source still applies.

Reply:
Do you have a todo list?

Reply:
I'll resolve conversation threads as they're done, like this one now is.